test/powershell/Modules/Microsoft.PowerShell.LocalAccounts/Pester.Command.Cmdlets.LocalAccounts.LocalGroup.Tests.ps1

```diff
@@ -0,0 +1,853 @@
+        $result.ObjectClass | Should Be Group
+
+        RemoveTestGroups -basename $nameMax
+    }
+
+    It "Errors on New-LocalGroup with name(256) over max" {
+        $name = "A"*257
+        $desc = "D"*129
+        $actualFqeid = ""
+
+        try {
+            $shouldBeNull = New-LocalGroup -Name $name -Description $desc
+
+            #clean up erroneous creation
+            if ($shouldBeNull) { Remove-LocalGroup -Name $name }
+        }
+        catch {
+            $actualFqeid = $_.FullyQualifiedErrorId 
+        }
+        $actualFqeid | Should Be "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.NewLocalGroupCommand"
+    }
+
+    It "Creates New-LocalGroup with Description > 48 characters" {
+        $descMax = "Test Group Add Description that is longer than 48 characters"
+        $result = New-LocalGroup -Name TestGroupAddRemove -Description $descMax
+
+        $result.Description | Should BeExactly $descMax
+    }
+
+    It "Errors on Invalid characters" {
+        #Arrange
+        #list of characters that should be invalid
+        $InvalidCharacters = @"
+\/"[]:|<>+=;,?*
+"@
+        $failedCharacters = @()
+
+        $InvalidCharacters = $InvalidCharacters[0..($InvalidCharacters.Length - 1)]
+
+        #Act
+        foreach ($character in $InvalidCharacters) {
+            try {
+                $invalidGroup = New-LocalGroup -Name ("InvalidBecauseOf" + $character) -ErrorAction Stop
+            }
+            catch {
+            }
+            finally {
+                if ($invalidGroup) {
+                    Remove-LocalGroup -Name $invalidGroup
+                    $failedCharacters += $character
+                }
+            }
+        }
+
+        if ($failedCharacters.Count -gt 0) { Write-Host "characters causing test fail: $failedCharacters" }
+        $failedCharacters.Count -eq 0 | Should Be true
+    }
+
+    It "Error on names containing only spaces" {
+        $sb = { New-LocalGroup -Name "   " }
+
+        VerifyFailingTest $sb "InvalidName,Microsoft.PowerShell.Commands.NewLocalGroupCommand"
+    }
+
+    It "Error on names containing only periods" {
+        $sb = { New-LocalGroup -Name "..." }
+
+        VerifyFailingTest $sb "InvalidName,Microsoft.PowerShell.Commands.NewLocalGroupCommand"
+    }
+
+    It "Errors on names ending in a period" {
+        $sb = {
+            New-LocalGroup -Name "TestEndInPeriod."
+        }
+
+        VerifyFailingTest $sb "InvalidName,Microsoft.PowerShell.Commands.NewLocalGroupCommand"
+
+        $sb = {
+            New-LocalGroup -Name ".TestEndIn.Period.."
+        }
+
+        VerifyFailingTest $sb "InvalidName,Microsoft.PowerShell.Commands.NewLocalGroupCommand"
+    }
+
+    It "Errors on Name over 256 characters" {
+        $sb = { New-LocalGroup -Name ("A"*257) }
+
+        VerifyFailingTest $sb "ParameterArgumentValidationError,Microsoft.PowerShell.Commands.NewLocalGroupCommand"
+    
+        RemoveTestGroups -basename ("A"*257)
```

Comment:
It's better to put `RemoveTestGroups` in a finally block. Or maybe it's not needed in this test